api/web3server_utils.go

```diff
@@ -117,9 +118,28 @@ func (svr *web3Handler) getTransactionFromActionInfo(blkHash hash.Hash256, selp
 	if err != nil {
 		return nil, err
 	}
-	tx, err := action.RawTxToSignedTx(ethTx, svr.coreService.EVMNetworkID(), selp.Signature())
-	if err != nil {
-		return nil, err
+	var (
+		tx       *types.Transaction
+		sig      = selp.Signature()
+		encoding = iotextypes.Encoding(selp.Encoding())
+	)
+	if encoding != iotextypes.Encoding_IOTEX_PROTOBUF {
+		signer, err := action.NewEthSigner(encoding, svr.coreService.EVMNetworkID())
+		if err != nil {
+			return nil, err
+		}
+		tx, err = action.RawTxToSignedTx(ethTx, signer, sig)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		if sig[64] >= 27 {
+			sig[64] -= 27
+		}
+		tx, err = ethTx.WithSignature(types.HomesteadSigner{}, sig)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
```
    if encoding != iotextypes.Encoding_IOTEX_PROTOBUF {
        signer, err = ...
    } else {
        signer = types.Homestead...
    }
    tx, err := action.RawTxToSignedTx(ethTx, signer, sig)
    if err != nil {
        return nil, err
    }
```
logic like `sig[64] -= 27` is something we want to hide in some unified util function. util function is help us to keep our logic clean.

Reply:
fixed